pkgs/development/go-modules/generic/default.nix

```diff
@@ -60,93 +60,107 @@ assert runVend != false -> throw "`runVend` has been replaced by `proxyVendor`";
 assert goPackagePath != "" -> throw "`goPackagePath` is not needed with `buildGoModule`";
 
 let
-  args = removeAttrs args' [ "overrideModAttrs" "vendorSha256" ];
-
-  go-modules = if vendorSha256 != null then stdenv.mkDerivation (let modArgs = {
-
-    name = "${name}-go-modules";
-
-    nativeBuildInputs = (args.nativeBuildInputs or []) ++ [ go git cacert ];
-
-    inherit (args) src;
-    inherit (go) GOOS GOARCH;
-
-    patches = args.patches or [];
-    patchFlags = args.patchFlags or [];
-    preBuild = args.preBuild or "";
-    sourceRoot = args.sourceRoot or "";
-
-    GO111MODULE = "on";
-
-    impureEnvVars = lib.fetchers.proxyImpureEnvVars ++ [
-      "GIT_PROXY_COMMAND" "SOCKS_SERVER"
-    ];
-
-    configurePhase = args.modConfigurePhase or ''
-      runHook preConfigure
-
-      export GOCACHE=$TMPDIR/go-cache
-      export GOPATH="$TMPDIR/go"
-      cd "${modRoot}"
-      runHook postConfigure
-    '';
+  package = stdenv.mkDerivation (finalAttrs: let
+    # Create local variables that coerce "null" to the default value for several
+    # attributes for use in subsequent expressions. This keeps those attributes
+    # "null" to avoid rebuilds.
```

Comment:
@zowoq i mean that if there's a choice between avoiding rebuilds and making something more maintainable then the maintainable choice should be chosen

Reply:
That's fair. If this moves forward and this is the general sentiment, I can encode those attributes explicitly as their defaults (e.g. boolean variables are `false` instead of `null`). Thanks for reviewing.